.gitignore

```diff
@@ -47,12 +47,14 @@ tmp*
 tests/*.unicode
 tests/unicode_test*
 *.txt
+!CMakeLists.txt
```

Comment:
There may be a topic about improving the `.gitignore` content,
but I would prefer to keep it separate from this topic (i.e. a different PR),
so that we can then focus the review on why the `.gitignore` would benefit an update.

Reply:
Shall i just revert and repush - or some rebase type thing (which is where I have previously created some unlikley tangles due to my newness with using git)?

Reply:
In theory, one should go back to the commit that made the unwanted change,
and either remove it from the chain if that's all it does,
or modify it to remove the change, if it's part of a larger commit, and rebase everything on top of it.

But playing with the block chain can be complex and error prone.
So at this point, it's also somewhat simpler to just erase the unwanted changes manually and make that a new commit.

What matter is that, as a whole, this multi-commits PR doesn't modify `.gitignore` (and we can have a separate PR for that).